applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -49,11 +57,76 @@ def SetUpModel(self):
             self.solver.ImportModelPart()
             self.solver.AddDofs()
 
-        # Fill a Model instance using input
-        self.model = Kratos.Model()
         self.model.AddModelPart(self.main_model_part)
 
-    def SetUpAuxiliaryProcesses(self):
+        # this should let eventual derived stages modify the model after reading.
+        self.ModifyInitialProperties()
+        self.ModifyInitialGeometry()
+
+        self._SetUpListOfProcesses()
+        self._SetUpAnalysis()
+
+    def RunSolutionLoop(self):
+
+        for process in self.list_of_processes:
+            process.ExecuteBeforeSolutionLoop()
+
+        while self.time < self.end_time:
+
+            self.time = self.solver.AdvanceInTime(self.time)
+            self.step = self.main_model_part.ProcessInfo[Kratos.STEP]
+
+            if self.is_printing_rank:
+                Kratos.Logger.PrintInfo("Fluid Dynamics Analysis","STEP = ", self.step)
```

Comment:
If you print `self.main_model_part.ProcessInfo[KratosMultiphysics.STEP]` then you don't have to save the step I think